### PR TITLE
make builds work and tests pass.

### PR DESCRIPTION
**Pull request by bangpound**
* updates ember and ember-cli dependencies.
* updates tests to use injected assert object.
* uses new names for imports.
* switch from deprecated Brocfile.js to ember-cli-build.js.
* switch from deprecated baseURL property in environment to rootURL.
* fixed a broken test that only worked outside DST or UTC.
* bump version.
* update travis-ci config.
  * tests against 6 named versions of ember

You can see the automated build output at https://travis-ci.org/bangpound/ember-datepicker/builds/296325228. Demo at https://bangpound.org/ember-datepicker/.

### CODE REVIEW

**Comment by bangpound** [comment]
Hey @ActiveCampaign/segment or @ActiveCampaign/contact should look at this — I think the component is used primarily there — and let me know if you have feedback. This has no urgency except it should be addressed before we make more modifications to this project because it cannot be tested without this PR.

I've tested the component with these changes in our app, so I'm reasonably confident that these changes won't cause regressions.

There are still some outstanding issues, though:

* Is the change here https://github.com/wesm87/ember-datepicker/pull/52/files#diff-d11e8c362837b42c08f445c2b32165c5R88 something we should include? This is some Ember.js stuff that I don't understand.
* There are deprecation notices related to `didInsertElement` which are possibly addressed in https://github.com/wesm87/ember-datepicker/pull/50. Is that how we should deal with it?

File: package.json
Comment (JFreemanActiveCampaign) [inline package.json:12]: If I'm understanding this correctly, this allows us to test this component against multiple versions of Ember. Is that so we can get a heads up that things are going to break, and we can take appropriate action before it ruins your weekend?

**Comment by cojamac** [comment]
@bangpound re outstanding issues.

* At one point ember modified the String prototype to include a `.fmt()` method that was somewhat analogous to sprintf. That was deprecated in favor of Es6 string interpolation. I believe we may have addressed that in our branch. I feel like we created this branch because we needed to eliminate deprecation warnings before upgrading to ember 2.0

* ~~I'm not sure the solution for the `didInsertElement` deprecation is actually an improvement. The "do not set inside X" warning have to do with known performance bottlenecks in resolving state changes. Using the `run.next()` addresses the deprecation but I feel like the point of the deprecation is to reduce the cascading effect of state changes. It may not be significant for the way we are using the datepicker. I'm going to try to get another opinion here.~~

**Comment by cojamac** [comment]
Gah! Somehow I was looking at the wrong commit for the didInsertElement. https://github.com/wesm87/ember-datepicker/pull/50/commits/380757a057025d2bc893cee19c216a1269352e51?w=1

I believe this is the correct diff to illustrate the fix.

**Comment by JFreemanActiveCampaign** [comment]
Everything in this looks good to me, really good. Echoing Collin's statement, he's bang on the money with the first point. In ES6, you can use template strings so
`` `${var1},${var2}` ``
is functionally equivalent to
`var1 + ',' + var2`

**Comment by cojamac** [comment]
I think both the outstanding fixes make sense. Moving the logic from `didInsertElement` into an `init` event makes sense. It would be marginally more correct to override the `init()` method instead use the `.on('init')` event... at least last time I checked. https://dockyard.com/blog/2015/10/19/2015-dont-dont-override-init

**Comment by bangpound** [comment]
this is built on top of @JFreemanActiveCampaign’s earlier work to fix the string interpolation. that hasn't changed in this branch.

File: package.json
Comment (bangpound) [reply package.json:12]: for two reasons:

* what you said
* and it's what new ember addon projects generated by the cli tool also do

if you check out https://travis-ci.org/bangpound/ember-datepicker/builds/212717647 you'll see that it produces results for each version, and we can also indicate that a version can safely fail without failing the whole build.

**Comment by bangpound** [comment]
I just noticed this deprecation warning showing up in the CI build:

> The `initialize` method for Application initializer \\\'App Version\\\' should take only one argument - `App`, an instance of an `Application`. [deprecation id: ember-application.app-initializer-initialize-arguments] 

I also see this in the browser console when running the app in debug mode on localhost. I actually do not understand why it's relevant here, so I would appreciate your advice.

**Comment by cojamac** [comment]
@bangpound https://github.com/ActiveCampaign/ember-app/blob/version-8.8/app/initializers/admin-service.js

We have a number of initializers in our ember app. Apparently the function signature for `initialize` changed a while back. Honestly all of these need to die in a fire. They account for a lot of the untestable magic on the platform. And being untestable magic, they are difficult to remove.

**Comment by bangpound** [comment]
I picked this PR up on the weekend because I built our Ember app and was again alarmed by the deprecation notices that remain. I am clearly not Ember smart… I tried to uodate the usage `didInsertElement` and replace it with an `init` hook and an `init` event, but the tests consistently failed when I did this. I do not think I can complete this without more specific guidance. 

On a positive note, I believe that when we start fixing some of the deprecation notices, the build happens way way more quickly in the main app. I saw the build time drop to a few seconds.

**Comment by AC-TimRourke** [comment]
@bangpound I'll trade you Behat thoughts for Ember thoughts if you want to poke at this later.

My hunch is that the PR changing the `didInsertElement` to manually scheduling work to be done in the `afterRender` queue is the correct move. What test failures are you seeing?

**Comment by bangpound** [comment]
@AC-TimRourke Literally every test fails. It's a "programmer doesn't get it" error.

**Comment by bangpound** [comment]
This is ready to review finally thanks to @AC-TimRourke's work. When this is merged, we can also add this repository to travis as an open source project and then we'll have automated tests running here.

@cojamac @JFreemanActiveCampaign I'm tagging you as the original reviewers. Finishing this work will allow us to improve testing in CircleCI on ember.

**Review by JFreemanActiveCampaign — APPROVED** [review]
I don't see any issues, but I am mostly trusting the config is correct, because I don't know the format well enough

**Comment by bangpound** [comment]
Sorry. There was a mixup. Tim targeted the master branch for his changes. I've merged those changes into develop which is where this PR is made.

Testing outcomes are here: https://travis-ci.org/bangpound/ember-datepicker/builds/296359727

All green! Let's go!

**Comment by subtext** [comment]
All of this looks good to me. I'm glad I was included on this thread by whatever means.

**Comment by AC-TimRourke** [comment]
@cojomac any thoughts?

**Comment by cojamac** [comment]
Looks good to me. I don't have write access though. @rhealitycheck @ac-pevans

**Pull request merged**
